lib/ExpressRedisCache/route.js

```diff
@@ -166,46 +175,39 @@ module.exports = (function () {
           /** if it's cached, display cache **/
 
           if ( cache.length ) {
-            res.contentType(cache[0].type || "text/html");
-            if(binary){ //Convert back to binary buffer
-              res.send(new Buffer(cache[0].body, 'base64'));
-            }else{
-              res.send(cache[0].body);
-            }
+            res.socket.write(cache[0].body);
+            res.end();
           }
 
           /** otherwise, cache request **/
           else {
+            var _write = res.socket.write.bind(res.socket);
 
-            /** wrap res.send **/
-            var send = res.send.bind(res);
-
-            res.send = function (body) {
-
+            res.socket.write = function redisCacheWrite (body) {
               /** send output to HTTP client **/
-              var ret = send(body);
+              _write.apply(_write, Array.prototype.slice.call(arguments));
 
               /** convert binary to base64 string **/
               if(binary && typeof body !== 'string'){
                 body = new Buffer(body).toString('base64');
               }
 
-              /** save only strings to cache **/
-              if ( typeof body !== 'string' ) {
-                return ret;
-              }
-
-              /** Create the new cache **/
-              self.add(name, body, {
-                  type: this._headers['content-type'],
-                  expire: expirationPolicy(res.statusCode)
-                },
-                domain.intercept(function (name, cache) {}));
-
-              return ret;
-
+              // add to the cache entry being built
+              buf += body;
             };
 
+            /** wait for response to finish streaming, then cache it **/
+            onFinished(res, function (err) {
+              if (!err) {
+                /** Create the new cache **/
+                self.add(name, buf, {
+                    type: res._headers['content-type'],
```

Comment:
This might not be required now, since the *content-type* is stored in the _body_